src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -59,27 +59,32 @@ static auto with(
 }
 
 template<typename Func>
-static auto with(
+static auto with_free(
   ss::shared_ptr<tm_stm> stm,
   kafka::transactional_id tx_id,
   const std::string_view name,
-  model::timeout_clock::duration timeout,
   Func&& func) noexcept {
-    return stm->get_tx_lock(tx_id)
-      ->with(
-        timeout,
-        [name, tx_id, func = std::forward<Func>(func)]() mutable {
-            vlog(txlog.trace, "got_lock name:{}, tx_id:{}", name, tx_id);
-            return ss::futurize_invoke(std::forward<Func>(func))
-              .finally([name, tx_id]() {
-                  vlog(
-                    txlog.trace,
-                    "released_lock name:{}, tx_id:{}",
-                    name,
-                    tx_id);
-              });
-        })
-      .finally([tx_id, stm]() { stm->try_rm_lock(tx_id); });
+    auto f = ss::now();
```

Comment:
I'm not sure I followed what this means..

> But if it waits (with timeout)
then there is a chance of a temporaty deadlock because try_abort locks
a data partition so the current holder of the lock on the coordinator
side may try interact with it and got locked

What is a temporary deadlock?

Reply:
```
data partition:        |         txn coordinator
-----------------------------------------------------
// tx expired          |    // tx is being committed at the same time
lock(tx.id)            |
                       |    lock(tx.id)
invoke try_abort       |
                       |    receives try_abort
                       |      lock(tx.id, timeout) <- this lock waits, so lock
                       |                              on data partition waits too
                       |    invoke commit
receives try_abort     |
lock(tx.id) <- waits   |
               for 1st |
               lock    |
```

timeout was controlling duration of the temporary deadlock, now instead of the timeout we check if the lock is ready and take it only in this case